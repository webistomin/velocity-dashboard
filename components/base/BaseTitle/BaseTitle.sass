+b('base-title')

  +m('level-1')
    font: var(--text-heading-1)

  +m('level-2')
    font: var(--text-heading-2)

  +m('level-3')
    font: var(--text-heading-3)

  +m('level-4')
    font: var(--text-heading-4)
