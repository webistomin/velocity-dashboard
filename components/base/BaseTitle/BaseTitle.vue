<template lang="pug" functional>
  component(
    v-bind='{is: `h${props.level}`}'
    :class='[`base-title_level-${props.level}`, data.staticClass, data.class]'
    ).title.base-title
    slot
</template>

<script lang="ts">
  import Vue from 'vue';

  export default Vue.extend({
    name: 'BaseTitle',
    props: {
      level: {
        type: Number,
        required: true,
        default: 1,
      },
    },
  });
</script>

<style lang="sass">
  @import "BaseTitle"
</style>
