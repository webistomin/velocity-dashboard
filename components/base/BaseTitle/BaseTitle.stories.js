import { number } from '@storybook/addon-knobs';
import BaseTitle from './BaseTitle';

export default {
  title: 'BaseTitle',
  parameters: {
    options: {
      showPanel: true,
      selectedPanel: 'storybook/knobs/panel',
      previewTabs: {
        actions: {
          hidden: true,
        },
      },
    },
  },
};

export const Default = () => ({
  props: {
    level: {
      default: number('Level', 1),
    },
  },
  render(h) {
    return h(
      BaseTitle,
      {
        props: {
          level: this.level || 1,
        },
      },
      [`Heading ${this.level || 1}`]
    );
  },
});
