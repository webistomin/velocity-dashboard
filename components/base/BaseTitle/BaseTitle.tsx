import { RenderContext, VNode } from 'vue';

import './BaseTitle.sass';

export interface IBaseTitleProps {
  level: number;
}

export const BaseTitle = (context: RenderContext<IBaseTitleProps>): VNode => {
  const { level } = context.props;
  const HeadingComponent = `h${level}`;
  return (
    <HeadingComponent
      class={`title base-title base-title_level-${level} ${context.data.staticClass || ''} ${context.data.class ||
        ''}`}>
      {context.children}
    </HeadingComponent>
  );
};
