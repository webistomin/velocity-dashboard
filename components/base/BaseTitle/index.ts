import { BaseTitle } from './BaseTitle';

export default BaseTitle;
