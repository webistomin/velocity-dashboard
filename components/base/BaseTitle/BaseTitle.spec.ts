import { mount } from '@vue/test-utils';
import { BaseTitle } from './BaseTitle';

describe('BaseTitle', () => {
  const factory = (propsData = {}, options = {}) => {
    return mount(BaseTitle, {
      propsData: {
        level: 1,
        ...propsData,
      },
      ...options,
    });
  };

  it('Matches snapshot', () => {
    const wrapper = factory().html();
    expect(wrapper).toMatchSnapshot();
  });
});
