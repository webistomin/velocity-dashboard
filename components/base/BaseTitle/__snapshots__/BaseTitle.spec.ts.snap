// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BaseTitle Matches snapshot 1`] = `"<h1 class=\\"title base-title base-title_level-1  \\"></h1>"`;
